docs/source/how_to/how_to_algorithm_selection.ipynb

```diff
@@ -35,34 +63,92 @@
    "source": [
     "import numpy as np\n",
     "\n",
-    "import optimagic as om"
+    "import optimagic as om\n",
+    "\n",
+    "\n",
+    "def trid_scalar(x):\n",
+    "    \"\"\"Implement Trid function: https://www.sfu.ca/~ssurjano/trid.html.\"\"\"\n",
+    "    return ((x - 1) ** 2).sum() - (x[1:] * x[:-1]).sum()\n",
+    "\n",
+    "\n",
+    "def trid_gradient(x):\n",
+    "    \"\"\"Calculate gradient of trid function.\"\"\"\n",
+    "    l1 = np.insert(x, 0, 0)\n",
+    "    l1 = np.delete(l1, [-1])\n",
+    "    l2 = np.append(x, 0)\n",
+    "    l2 = np.delete(l2, [0])\n",
+    "    return 2 * (x - 1) - l1 - l2"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "@om.mark.least_squares\n",
-    "def sphere(params):\n",
-    "    return params\n",
+    "## Step 1: Theory\n",
+    "\n",
+    "The below decision tree offers a practical guide on how to narrow down the set of algorithms to experiment with, based on the theoretical properties of your problem:\n",
```

Comment:
```suggestion
    "This is a practical guide for narrowing down the set of algorithms to experiment with:\n",
```

Actually I would add the graph above the heading "Example problem". So you see the four steps, but also have the decision tree. It is more general than the example, all the rest is only relevant in there.

Below the graph, add an explanation on why the optimisers in a node are the relevant ones. How would I  be able to complete the "..." otherwise?

Reply:
> Going in this direction: Can we link the recommended algorithms to the optimizer 'types' mentioned in explanation_of_numerical_optimizers.ipynb? 

No, the types are unrelated. Maybe we should make this message clearer! 

Understanding the algorithm types mostly becomes relevant once you think about configuring an algorithm.